apps/api/v2/src/ee/calendars/controllers/calendars.controller.ts

```diff
@@ -55,4 +80,102 @@ export class CalendarsController {
       data: calendars,
     };
   }
+
+  @UseGuards(AccessTokenGuard)
+  @Get("/office365/connect")
```

Comment:
As I understand, gcal and microsoft outlook have the same enpoints - getting redirect url, save and check.

First, why do we call `@Get("/office365/connect")` here but in `gcal.controller.ts` its `@Get("/oauth/auth-url")` ?

Next, since we will have 1 calendars endpoint and ultimately get rid of gcal controller (as i remember our discussion), we want to re-use route logic for google calendar, microsoft outlook and other calendars that follow this structure.

Secondly, that means that we need a dynamic route `@Get("/:calendar/connect")` that then based on `:calendar` be it google or microsoft outlook invokes their corresponding services be it `GCalService` or `MicrosoftOutlookService`.

Also, we can define an interface `CalendarApp` foe example that looks like (can probably have different function parameters but this is just example):
```
interface CalendarApp {
  redirect(authorization: string, req: Request): Promise<{
    status: string;
    data: {
      authUrl: string;
    };
  }>;

  save(state: string, code: string): Promise<{
    url: string;
  }>;

  check(userId: number): Promise<{
    status: string;
  }>;
}
```

Then, the MicrosoftOutlookService and GCalService will implement it.